lib/ui/src/components/sidebar/SidebarItem.tsx

```diff
@@ -96,15 +105,22 @@ export const Item = styled(({ className, children, id }) => (
     }
 );
 
-export default function SidebarItem({
-  name,
-  isComponent,
-  isLeaf,
-  isExpanded,
-  isSelected,
+type SidebarItemProps = React.ComponentProps<typeof Item> & {
+  isComponent?: boolean;
+  isLeaf?: boolean;
+  isExpanded?: boolean;
+  isSelected?: boolean;
+};
+
+const SidebarItem = ({
+  name = 'loading story',
+  isComponent = false,
+  isLeaf = false,
+  isExpanded = false,
+  isSelected = false,
   ...props
-}) {
-  let iconName;
+}: SidebarItemProps) => {
+  let iconName: React.ComponentProps<typeof Icons>['icon'];
```

Comment:
Is this the right way to do this? Wondering if it's better to export a type? This looks a little verbose to me, but I'm a typescript newb.

Reply:
The top-level `@storybook/components` isn't directly expose the `IconsProps`, which makes import path a bit hard to find (need to dig where the type is defined in the `node_module`)... So I just use the helper type from React to extract the `IconsProps` (`IconsProps === React.ComponentProps<typeof Icons>`).